lib/ansible/modules/cloud/hcloud/hcloud_server.py

```diff
@@ -287,9 +295,10 @@ def _update_server(self):
             server_type is not None
             and self.hcloud_server.server_type.name != server_type
         ):
+            state = self.module.params.get("state")
             if self.hcloud_server.status == Server.STATUS_RUNNING:
                 if not self.module.check_mode:
-                    if self.module.params.get("force_upgrade"):
+                    if self.module.params.get("force_upgrade") or state == "stopped":
```

Comment:
Remember the state before stop so it can be latter used.
~~~diff
                      if self.module.params.get("force_upgrade") or state == "stopped":
+                         previous_state = self.hcloud_server.status
~~~